### PR TITLE
Start of 404 and 426 implementation for getTitles

### PR DESCRIPTION
**Pull request by swong15**
This pull request begins implementation of the 404 (no titles available) and 426 (old commit, client should commit suicide) status codes in getTitle.

Old commit logic by putting an old TID into commits table and 404 tested by emptying out local Cassandra data store.

### CODE REVIEW

**Comment by swong15** [comment]
gwicke/testreduce#16

File: CassandraBackend.js [hunk 1]
```diff
@@ -195,19 +209,23 @@ function getTestToRetry() {
...
  * JSON, for example [ 'enwiki', 'some title', 12345 ]
  */
-CassandraBackend.prototype.getTest = function (commit, cb) {
-    var retry = (getTestToRetry.bind(this))();
-    if (retry) {
-        return retry;
+CassandraBackend.prototype.getTest = function (clientCommit, clientDate, cb) {
+    var retry = this.getTestToRetry(),
+        lastCommitTimestamp = this.commits[0].timestamp,
+        retVal = 404;
```
Comment (gwicke) [inline CassandraBackend.js]: Somehow directly returning HTTP status codes from the backend feels wrong to me. Maybe we can re-define getTest to return an object with an 'error' member instead? Something like this:
{ error: { code: 404, message: 'No tests to run for this commit' } } // error case
{ error: { code: 426, message: 'Commit too old' } } // error case
{ test: '{prefix:....}'} // no error

But really we should probably use names:
{ error: { code: 'ResourceNotFoundError', message: 'No tests to run for this commit' } } // error case
{ error: { code: 'BadCommitError', message: 'Commit too old' } } // error case

According to http://tools.ietf.org/html/draft-ietf-httpbis-p2-semantics-16#page-34 426 actually means something else. Code 400 (Bad Request) seems to be more accurate.

Comment (gwicke) [inline CassandraBackend.js]: This would be somewhat similar to http://mcavage.me/node-restify/#Error-handling

Comment (swong15) [inline CassandraBackend.js]: The parsoid client currently accepts 426 as the suicide status code. That logic will have to be changed in the client at some point.

**Pull request merged**